### PR TITLE
Fix compilation on LLVM svn

### PR DESCRIPTION
**Pull request by yuyichao**
Ref llvm-mirror/llvm@20a600c
Ref https://github.com/llvm-mirror/llvm/commit/faaafe58c6aaa5c523892eaf5b65fc642097513c

### CODE REVIEW

File: src/codegen.cpp [hunk 1]
```diff
@@ -4422,6 +4422,9 @@ static std::unique_ptr<Module> emit_function(
         // TODO: Fix when moving to new LLVM version
         #if JL_LLVM_VERSION < 30400
         dbuilder.createCompileUnit(0x01, filename, ".", "julia", true, "", 0);
+        #elif JL_LLVM_VERSION >= 40000
```
Comment (tkelman) [inline src/codegen.cpp]: we're a little inconsistent about whether or not we indent ifdefs - most of the time it looks like we don't though

Comment (yuyichao) [inline src/codegen.cpp]: I do like no indent better and I'm just making it consistent with whatever around it.... Should I change this?

**Comment by yuyichao** [comment]
All tests passed with llvm SVN locally (x64).

**Review by tkelman — APPROVED** [review]
these were before 4.0 branched, right? is trunk 4.1 yet?

**Comment by yuyichao** [comment]
Right, I believe these were in Dec 2016 before 4.0 is branched. Trunk is 5.0 now.

**Comment by tkelman** [comment]
ah right they're using major versions like gcc now. is our macro robust to when they hit v10?

**Comment by yuyichao** [comment]
The macro should be robust until they hit `n.100.0` or `214748.36.48`, whichever is earlier ;-p

**Pull request merged**